pkg/sync/sync_context.go

```diff
@@ -829,6 +827,24 @@ func (sc *syncContext) autoCreateNamespace(tasks syncTasks) syncTasks {
 	return tasks
 }
 
+func (sc *syncContext) appendNsTask(tasks syncTasks, preTask *syncTask, unstructuredObj *unstructured.Unstructured) syncTasks {
+	modified, err := sc.namespaceModifier(unstructuredObj)
+	if err != nil {
+		tasks = sc.appendFailedNsTask(tasks, unstructuredObj, err)
+	} else if modified {
+		tasks = append(tasks, preTask)
+	}
+
+	return tasks
+}
+
+func (sc *syncContext) appendFailedNsTask(tasks syncTasks, unstructuredObj *unstructured.Unstructured, err error) syncTasks {
+	task := &syncTask{phase: common.SyncPhasePreSync, targetObj: unstructuredObj}
+	sc.setResourceResult(task, common.ResultCodeSyncFailed, common.OperationError, fmt.Sprintf("Namespace auto creation failed: error retrieving namespace from k8s: %s", err))
```

Comment:
As `appendFailedNsTask` can be called in different places this error message can be misleading. The error message needs to be added to the `err` previously.